### ISSUE TITLE
geom_stats() not shown

### ISSUE THREAD

**Issue by mikejiang** [RGLab/ggcyto#15]
On 01/30/2017 07:54 AM, Franziska Witzel wrote:

> Hello!
>
> Thank you very much for ggcyto!!!!
>
> I have a flowset for which I have created a 1D rectangular gate, like this
>
>  posgate <- rectangleGate("FL4.A"=c(1e3,Inf),filterId = "pos")
>
> Then I replicate this gate for each frame in my flowset.
>
> posgates <- sapply(sampleNames(egfset), function(x) posgate)
>
> I want to draw the data AND the gate AND also show the stats:
>
> ggcyto(egfset,aes(x=FL4.A)) + geom_density() + geom_gate(posgates) + 
>   geom_stats() + scale_x_flowJo_biexp()
>
> However, the stats always get removed from view, as also indicated by the 
> warning (egfset contains 5 flow frames): 
>
> Removed 5 rows containing missing values (geom_label).
>
> I saw that one other person had posted this issue on github and it was caused by NA
> values in the data. However, firstly, I have not transformed my data when reading them in.
> My frames contain a lot of zero values, but no negative values. The gate checks out "ok",
> because it can be used for subsetting the data successfully. It is also drawn correctly with
> geom_gate(posgates).
> As I could not reproduce this problem with data from here:  data(GvHD), I guess it must be
> related to my data. Do you have any other ideas/suggestions about what might cause this issue?
>

**Comment by mikejiang** [comment]
Indeed, your data somehow has exaggerated measurement range (compared to the actual data range).
```r
> egfset[[1]]
flowFrame object 'A06_solventCt.fcs'
with 26461 cells and 14 observables:
      name  desc    range minRange maxRange
$P1  FSC.A FSC-A 16777216        0 16777215
$P2  SSC.A SSC-A 16777216        0 16777215
$P3  FL1.A FL1-A 16777216        0 16777215
$P4  FL2.A FL2-A 16777216        0 16777215
$P5  FL3.A FL3-A 16777216        0 16777215
$P6  FL4.A FL4-A 16777216        0 16777215
```
```r
> range(egfset[[1]], type = "data")
       FSC.A   SSC.A FL1.A FL2.A FL3.A  FL4.A   FSC.H   SSC.H FL1.H FL2.H FL3.H  FL4.H Width Time
min  2326916  102522     0     0     0      0 1255702   95444    81     0     0    137    73  155
max 12826179 1992038 90881 36056 48274 325524 9779722 9188389 24631  8763 12380 296188   390 2344
```
`geom_stats` computes the `stats` location based on this `instrument range` by default, which is why you won't see it on the plot. 
The solution is to explicitly supply the `data_range` with the correct values
```r
rg <- range(egfset[[1]], type = "data")
p + geom_stats(data_range = rg) 
```
Note that `range(x, type = "data")` method is only available in the latest `flowCore 1.41.5` , if you don't want to update it,  you can do it manually:
```r
rg <- data.frame(apply(exprs(egfset[[1]]), 2, range))
```
![image](https://cloud.githubusercontent.com/assets/1385649/22485604/2182736c-e7bb-11e6-97c8-a101787d7d1a.png)

**Comment by mikejiang** [comment]
There was some debate on which range should be the default.  The actual `data range` is not always desirable when the extreme outlier events are present,  e.g.
```r
autoplot(gs, "CD8") + ggcyto_par_set(limits = "data")
```
![image](https://cloud.githubusercontent.com/assets/1385649/22799732/b0c89798-eebb-11e6-99b4-8fd2a3d2209f.png)
we have seen the case where the outliers are even more far left and Imagine instead of `ellipseGate`, if we have a `rectangelGate` here that is intended to cover the whole top left quadrant , using the actual `data range` will push the 2D density far right and `stats` label to far left thus the plots won't be visually helpful.
